src/lsdb/catalog/catalog.py

```diff
@@ -248,6 +251,31 @@ def cone_search(self, ra: float, dec: float, radius: float):
         ddf_partition_map = {pixel: i for i, pixel in enumerate(pixels_in_cone)}
         return Catalog(cone_search_ddf, ddf_partition_map, filtered_hc_structure)
 
+    def polygon_search(self, polygon: SingleSphericalPolygon) -> Catalog:
+        """Perform a polygonal search to filter the catalog.
+
+        Filters to points within the polygonal region specified in ra and dec, in degrees.
+        Filters partitions in the catalog to those that have some overlap with the region.
+
+        Args:
+            polygon (SingleSphericalPolygon): The polygon to filter pixels with.
+
+        Returns:
+            A new catalog containing the points filtered to those within the
+            polygonal region, and the partitions that have some overlap with it.
+        """
+        filtered_hc_structure = self.hc_structure.filter_by_polygon(polygon)
+        pixels_in_polygon = filtered_hc_structure.get_healpix_pixels()
+        partitions = self._ddf.to_delayed()
```

Comment:
There's actually now a `self.to_delayed()` that can be used here.

Reply:
If you copied this from the cone filter then I forgot to update it there, mb

Reply:
Got it, will update both!